### DISCUSSION TITLE
Responses with `Content-Type: application/pdf` don't work in dev mode

### DISCUSSION THREAD

**Discussion by mattiaz9** [vercel/next.js#31980] [Help]
### What version of Next.js are you using?

12.0.4

### What version of Node.js are you using?

16

### What browser are you using?

Chrome, Safari

### What operating system are you using?

macOS

### How are you deploying your application?

next dev 

### Describe the Bug

I have a route where I generate a pdf from a component and return a pdf response. 
When I run the project with `next dev` the page is stuck on the initial loading as if the browser doesn't get any response, but when I run `next build && next start` everything works fine.

This is my code for the getServerSideProps method:
```
export const getServerSideProps: GetServerSideProps<any, { id: string }> = async ({ params, req, res }) => {
  const { id } = params!
  const recipe = await adminFetchRecipe(id)

  const pdf = await componentToPdf(
    <NutritionLabel
      nutrition={recipe.nutrients}
      servings={recipe.servings ?? 1}
      servingSize={recipe.serving_size}
    />
  )

  res.setHeader("content-disposition", `attachment; filename="${slugify(recipe.name)}_label.pdf";`)
  res.setHeader("content-type", "application/pdf")
  res.write(pdf)
  res.end()

  return {
    props: {},
  }
}
```

I don't have this problem with other type of contents, like `text/plain` or `text/xml`

### Expected Behavior

The expected behaviour should be the same of the `next start` where the pdf file is correctly download from the browser

### To Reproduce

It can be reproduced by just settings the content-type to application/pdf in the getServerSideProps method like:
```
...
res.setHeader("content-type", "application/pdf")
res.write("")
res.end()

return {
  props: {},
}
...
```

**Comment by balazsorban44** [comment]
Not sure why this isn't an API route instead. I would put that logic under `/api`.
